erpnext/erpnext_integrations/doctype/authorizenet_settings/authorizenet_settings.py

```diff
@@ -0,0 +1,215 @@
+	payment = apicontractsv1.paymentType()
+	payment.creditCard = credit_card
+
+	pr = frappe.get_doc("Payment Request", data.reference_docname)
+	sales_order = frappe.get_doc("Sales Order", pr.reference_name).as_dict()
+
+	customer_address = apicontractsv1.customerAddressType()
+	customer_address.firstName = data.payer_name
+	customer_address.address = sales_order.customer_address[:60]
+
+	# Create order information 
+	order = apicontractsv1.orderType()
+	order.invoiceNumber = sales_order.name
+
+	for item in sales_order.get("items"):
+		for i in range(len(sales_order.get("items"))):
+
+			# setup individual line items
+			item[i] = apicontractsv1.lineItemType()
+			item[i].itemId = item.item_code
+			item[i].name = item.item_name[:30]
+			item[i].description = item.item_name
+			item[i].quantity = item.qty
+			item[i].unitPrice = item.amount
+
+			# build the array of line items
+			line_items = apicontractsv1.ArrayOfLineItem()
+			line_items.lineItem.append(item[i])
+
+	# Create a transactionRequestType object and add the previous objects to it.
+	transaction_request = apicontractsv1.transactionRequestType()
+	transaction_request.transactionType = "authCaptureTransaction"
+	transaction_request.amount = data.amount
+	transaction_request.payment = payment
+	transaction_request.order = order
+	transaction_request.billTo = customer_address
+	transaction_request.lineItems = line_items
+
+	# Assemble the complete transaction request
+	create_transaction_request = apicontractsv1.createTransactionRequest()
+	create_transaction_request.merchantAuthentication = merchant_auth
+	create_transaction_request.transactionRequest = transaction_request
+
+	# Create the controller
+	createtransactioncontroller = createTransactionController(
+		create_transaction_request)
+	createtransactioncontroller.execute()
+
+	response = createtransactioncontroller.getresponse()
+
+	if response is not None:
+		# Check to see if the API request was successfully received and acted upon
+		if response.messages.resultCode == "Ok":
+			# Since the API request was successful, look for a transaction response
+			# and parse it to display the results of authorizing the card
+			if hasattr(response.transactionResponse, 'messages') is True:
+				status = "Completed"
+			else:
+				status = "Failed"
+				if hasattr(response.transactionResponse, 'errors') is True:
+					status = "Failed"
+
+		# Or, print errors if the API request wasn't successful
+		else:
+			status = "Failed"
+			if hasattr(response, 'transactionResponse') is True and hasattr(
+					response.transactionResponse, 'errors') is True:
+				status = "Failed"
+
+			else:
+				status = "Failed"
+
+	custom_redirect_to = None
+
+	if status != "Failed":
+		try:
+			custom_redirect_to = frappe.get_doc(data.reference_doctype, data.reference_docname).run_method("on_payment_authorized",
+				status)
+		except Exception as ex:
+			raise ex
+
+	response = to_dict(response)
+	transId = response.get("transactionResponse").get("transId")
```

Comment:
Can we not just do `transId = response.transactionResponse.transId`?

Authorizenet's documentation seems to use that